app/controllers/api/v0/articles_controller.rb

```diff
@@ -1,9 +1,9 @@
 module Api
   module V0
     class ArticlesController < ApiController
-      before_action :authenticate_with_api_key_or_current_user!, only: %i[create update]
```

Comment:
General question, since `authenticate!` checks doorkeeper plus API key plus current_user, why cant we just get rid of this method `authenticate_with_api_key_or_current_user ` and use `authenticate!` everywhere in the API?

Reply:
I think for a time we felt most assured using `authenticate_with_api_key_or_current_user` will distinguish between what OAuth apps can or cannot have access to. It's also because anything OAuth app have access to will need scope checks and we didn't have that figured out yet. This is just my assumption though.

Reply:
For context: In the past, we had things that were only meant for internal consumption in the public API, so I believe this was added to account for that. However, as of #6255, this is no longer true, so maybe you want to investigate a bit and potentially simplify this @maestromac

Reply:
Agree with @maestromac , it's for restricting access when a user is authorized with the doorkeeper_token for the time while we only have basic scopes.
As I see, we still don't have doorkeeper scopes for all the api features, so we may want to keep limiting the access in this case.

Reply:
Anyway, I would keep the possible changes for another pr :)

Reply:
I think too we should err on the side of caution on opening up OAuth features to registered apps.